### PR TITLE
[Java][Client] Add PATCH method support for jersey2

### PR DESCRIPTION
**Pull request by hisener**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: master for non-breaking changes and `3.0.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Add HTTP PATCH method for jersey2

### CODE REVIEW

**Comment by wing328** [comment]
@hisener thanks for the PR. Last time when we checked, Jersey 2.x didn't support "PATCH": https://github.com/swagger-api/swagger-codegen/issues/1924#issuecomment-173433123

Is it correct to say that the latest version of Jersey 2.x supports PATCH?

**Comment by hisener** [comment]
@wing328 Not actually. We achieve PATCH with workaround solution (HttpUrlConnectorProvider.SET_METHOD_WORKAROUND). I think it would be better than POST with X-HTTP-Method-Override header.

File: samples/client/petstore/java/jersey2-java6/src/main/java/io/swagger/client/ApiClient.java [hunk 1]
```diff
@@ -432,13 +433,12 @@ public String parameterToString(Object param) {
...
    *   APPLICATION/JSON
    *   application/vnd.company+json
-   * "* / *" is also default to JSON
    * @param mime MIME
    * @return True if the MIME type is JSON
    */
   public boolean isJsonMime(String mime) {
     String jsonMime = "(?i)^(application/json|[^;/ \t]+/[^;/ \t]+[+]json)[ \t]*(;.*)?$";
-    return mime != null && (mime.matches(jsonMime) || mime.equals("*/*"));
+    return mime != null && (mime.matches(jsonMime) || mime.equalsIgnoreCase("application/json-patch+json"));
```
Comment (wing328) [inline samples/client/petstore/java/jersey2-java6/src/main/java/io/swagger/client/ApiClient.java]: Please rebase on the latest master.

Comment (hisener) [inline samples/client/petstore/java/jersey2-java6/src/main/java/io/swagger/client/ApiClient.java]: Rebased but same diff. I think you didn't change `ApiClient.mustache` under `Java` with this commit https://github.com/swagger-api/swagger-codegen/commit/927055a681fb405eb202577cd6b0c6805662e3b8. Only under `Java/libraries` folder

Comment (wing328) [inline samples/client/petstore/java/jersey2-java6/src/main/java/io/swagger/client/ApiClient.java]: @hisener ok. I'll review that one later.

File: samples/client/petstore/java/jersey2/src/main/java/io/swagger/client/model/Cat.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public Cat declawed(Boolean declawed) {
    * @return declawed
   **/
   @ApiModelProperty(value = "")
-  public Boolean isDeclawed() {
+  public Boolean getDeclawed() {
```
Comment (wing328) [inline samples/client/petstore/java/jersey2/src/main/java/io/swagger/client/model/Cat.java]: @hisener please pull the latest as this should be changed to `isDeclawed` to follow Java boolean getter naming convention.

**Pull request merged**